doc-design/source/softwareArchitecture.rst

```diff
@@ -1188,8 +1189,14 @@ QSS performance considerations:
   derivatives (computing the necessary subset of the Jacobian) can be supported in the near term
   that would make their use with QSS more practical.
 
-- Longer term, automatic differentiation with atomic 2\ :sup:`nd` and 3\ :sup:`rd` derivative calls
+- Longer term, automatic differentiation with atomic 2\ :sup:`nd` and 3\ :sup:`rd` time-derivative calls
   will be valuable for efficient QSS solutions.
+  [Stuart: clarify what order you need for the state derivatives, and what order for the time
+  derivative of zero-crossing functions.]
```

Comment:
To do 3rd order QSS methods efficiently we would want 3rd derivatives of state and zero-crossing variables (so 2nd derivatives of state derivatives). And going up one order for each higher order QSS methods we want to implement.

With directional derivatives (putting performance aside) we have 2nd derivatives of state variables and only 1st derivatives of zero-crossing variables, so we are resorting to numeric differentiation of zero-crossing functions for 2nd order QSS and to do 3rd order QSS we'd have to do another level of numeric differentiation.